lib/datasource/galaxy/index.ts

```diff
@@ -1,71 +1,75 @@
 import { logger } from '../../logger';
-import { ExternalHostError } from '../../types/errors/external-host-error';
-import * as packageCache from '../../util/cache/package';
-import { Http } from '../../util/http';
+import { cache } from '../../util/cache/package/decorator';
+import { HttpResponse } from '../../util/http';
+import { Datasource } from '../datasource';
 import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import type { GalaxyResult } from './types';
 
-export const id = 'galaxy';
-export const defaultRegistryUrls = ['https://galaxy.ansible.com/'];
-export const customRegistrySupport = false;
-
-const http = new Http(id);
-
-export async function getReleases({
-  lookupName,
-  registryUrl,
-}: GetReleasesConfig): Promise<ReleaseResult | null> {
-  const cacheNamespace = 'datasource-galaxy';
-  const cacheKey = lookupName;
-  const cachedResult = await packageCache.get<ReleaseResult>(
-    cacheNamespace,
-    cacheKey
-  );
-  // istanbul ignore if
-  if (cachedResult) {
-    return cachedResult;
+export class GalaxyDatasource extends Datasource {
+  static readonly id = 'galaxy';
+
+  constructor() {
+    super(GalaxyDatasource.id);
   }
 
-  const lookUp = lookupName.split('.');
-  const userName = lookUp[0];
-  const projectName = lookUp[1];
-
-  const galaxyAPIUrl =
-    registryUrl +
-    'api/v1/roles/?owner__username=' +
-    userName +
-    '&name=' +
-    projectName;
-  const galaxyProjectUrl = registryUrl + userName + '/' + projectName;
-  try {
-    let res: any = await http.get(galaxyAPIUrl);
-    if (!res || !res.body) {
+  readonly customRegistrySupport = false;
+
+  readonly defaultRegistryUrls = ['https://galaxy.ansible.com/'];
+
+  @cache({
+    namespace: 'datasource-galaxy',
+    key: (getReleasesConfig: GetReleasesConfig) => getReleasesConfig.lookupName,
```

Comment:
```suggestion
    key: ({ lookupName }: GetReleasesConfig) => lookupName,
```
Just two small things. 😉